### PR TITLE
LPS-139658 | master

### PR DESCRIPTION
**Pull request by ozysouza**
https://issues.liferay.com/browse/LPS-139658

cc @david-gutierrez-mesa

### CODE REVIEW

**Comment by ozysouza** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/Forms.testcase [hunk 1]
```diff
@@ -324,6 +324,63 @@ definition {
 			rowEntry = "Created Form Name");
 	}
 
+	@description = "LPS-72048 - Verify that Belongs To rule can be paired with Guest role"
```
Comment (david-gutierrez-mesa) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/Forms.testcase:327]: Please use the standard way for description. example "This is a use case for LPS-XXXXXX. Test case description."

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/Forms.testcase [hunk 2]
```diff
@@ -1489,6 +1546,56 @@ definition {
 		LexiconEntry.viewActiveDisplayStyle(displayStyle = "table");
 	}
 
+	@description = "This is a use case for LPS-81578."
```
Comment (david-gutierrez-mesa) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/Forms.testcase:1549]: Please add test case description

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/Forms.testcase [hunk 3]
```diff
@@ -324,6 +324,63 @@ definition {
...
+			fieldLabel = "Text",
+			fieldName = "Text");
+
+		User.logoutPG();
+
+		Navigator.openSpecificURL(url = "${publishedFormPublicURL}");
+
+		FormFields.viewTextLabel(
+			fieldLabel = "Text",
+			fieldName = "Text");
```
Comment (david-gutierrez-mesa) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/Forms.testcase:381]: Please use tasks to separate tests steps

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/fields/FormsTextField.testcase [hunk 4]
```diff
@@ -1375,6 +1375,48 @@ definition {
 			fieldValue = "Teste");
 	}
 
+	@description = "LPS-84111 - Verify that Text Field can be autosaved on form builder"
```
Comment (david-gutierrez-mesa) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/fields/FormsTextField.testcase:1378]: Please use the standard way for description. example "This is a use case for LPS-XXXXXX. Test case description."

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/Forms.testcase [hunk 5]
```diff
@@ -1489,6 +1546,56 @@ definition {
...
+			publishedFormPublicURL = "${publishedFormPublicURL}"
+		);
+
+		Navigator.openSpecificURL(url = "${publishedFormPublicURL}");
+
+		User.logoutPG();
+
+		Navigator.openSpecificURL(url = "${publishedFormPublicURL}");
+
+		Alert.viewWarningSpecific(warningMessage = "You do not have the permission to view this form.");
```
Comment (david-gutierrez-mesa) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/Forms.testcase:1596]: Please use tasks to separate tests steps

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/fields/FormsTextField.testcase [hunk 6]
```diff
@@ -1375,6 +1375,48 @@ definition {
...
+
+		FormsAdminNavigator.gotoEditFormByName();
+
+		AssertElementPresent(
+			key_name = "First Field",
+			locator1 = "FormFields#FIELD_TARGET_LABEL");
+
+		AssertElementPresent(
+			key_name = "For Autosave",
+			locator1 = "FormFields#FIELD_TARGET_LABEL");
```
Comment (david-gutierrez-mesa) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/fields/FormsTextField.testcase:1417]: Please use tasks to separate tests steps

**Review by david-gutierrez-mesa — CHANGES_REQUESTED** [review]
Please take a look to the my commnets

**Comment by david-gutierrez-mesa** [comment]
Just started reviewing :)

**Comment by david-gutierrez-mesa** [comment]
Rebased. See #195

**Pull request closed without merging**